### PR TITLE
Fix some spelling typos

### PR DESCRIPTION
**Pull request by bwignall**
All changes should be non-semantic, spelling-only fixes.

### CODE REVIEW

**Comment by justinmk** [comment]
Thanks. But please undo the changes to all files in `/runtime/`, except tutor.
We don't maintain those files, and changing them makes it difficult to merge
them from Vim.

File: test/functional/api/mark_extended_spec.lua [hunk 1]
```diff
@@ -507,7 +507,7 @@ describe('API/extmarks', function()
     feed('0l3dl<esc>')
     check_undo_redo(ns, marks[1], 0, 2, 0, 1)
     check_undo_redo(ns, marks[2], 0, 3, 0, 1)
-    -- delete 1, nothing should happend to our marks
+    -- delete 1, nothing should happened to our marks
```
Comment (paulburlumi) [inline test/functional/api/mark_extended_spec.lua]: happen reads better here than happened.
```suggestion
    -- delete 1, nothing should happen to our marks
```

Comment (bwignall) [reply test/functional/api/mark_extended_spec.lua]: I will change to your suggestion.

**Comment by bwignall** [comment]
@justinmk , I will revert those and push momentarily. Sorry for my confusion.

File: test/functional/api/mark_extended_spec.lua [hunk 2]
```diff
@@ -543,7 +543,7 @@ describe('API/extmarks', function()
     check_undo_redo(ns, marks[1], 0, 1, 0, 0)
     check_undo_redo(ns, marks[2], 0, 3, 0, 0)
     check_undo_redo(ns, marks[3], 1, 2, 1, 0)
-    -- delete 1, nothing should happend to our marks
+    -- delete 1, nothing should happened to our marks
```
Comment (justinmk) [inline test/functional/api/mark_extended_spec.lua]:
```suggestion
    -- delete 1, nothing should happen to our marks
```

**Comment by blueyed** [comment]
@bwignall 
Thanks for this.
You might want to send the runtime ones to Vim.

**Pull request closed without merging**